internal/handlers/publishers.go

```diff
@@ -79,96 +86,132 @@ func (p *Publisher) GetPublisher(ctx *fiber.Ctx) error {
 
 // PostPublisher creates a new publisher.
 func (p *Publisher) PostPublisher(ctx *fiber.Ctx) error {
-	request := common.Publisher{}
+	normalize := normalizer.NewNormalizer()
+	request := new(common.Publisher)
 
-	if err := ctx.BodyParser(&request); err != nil {
-		return common.Error(fiber.StatusBadRequest, "can't create Publisher", "invalid json")
+	err := common.ValidateRequestEntity(ctx, request, "can't create Publisher")
+	if err != nil {
+		return err //nolint:wrapcheck
 	}
 
-	if err := common.ValidateStruct(&request); err != nil {
-		return common.ErrorWithValidationErrors(
-			fiber.StatusUnprocessableEntity, "can't create Publisher", "invalid format", err,
-		)
+	publisher := &models.Publisher{}
+
+	normalizedEmail := normalize.Normalize(request.Email)
+
+	publisher.ID = utils.UUIDv4()
+	publisher.Email = normalizedEmail
+
+	if request.ExternalCode != "" {
```

Comment:
These checks (the one below this too) can be omitted because [gorm won't save zero values](https://github.com/italia/developers-italia-api/pull/136).

Reply:
Unfortunately, if we remove this, the unique constraint will fail.